static/src/javascripts/projects/commercial/modules/messenger.ts

```diff
@@ -157,17 +158,30 @@ const toStandardMessage = (
 });
 
 /**
- * Retrieve a reference to the calling iframe via it's ID.
+ * Retrieve a reference to the calling iFrame
  *
- * Incoming messages contain the ID of the iframe into which the source window is embedded.
+ * Attempts the following strategies to find the correct iframe:
+ * - using the slotId from the incoming message
+ * - using the iframeId from the incoming message
+ * - checking message event.source (i.e. window) against all page level iframe contentWindows
+ *
+ * Listeners can use then use the iFrame to determine the slot making the postMessage call
  */
-const getIframe = (message: StandardMessage): HTMLIFrameElement | undefined => {
+const getIframe = (
+	message: StandardMessage,
+	messageEventSouce: MessageEventSource | null,
+): HTMLIFrameElement | undefined => {
 	if (message.slotId) {
 		const container = document.getElementById(`dfp-ad--${message.slotId}`);
 		return container?.querySelector('iframe') ?? undefined;
 	} else if (message.iframeId) {
 		const el = document.getElementById(message.iframeId);
 		return el instanceof HTMLIFrameElement ? el : undefined;
+	} else if (messageEventSouce) {
+		const iframes = document.querySelectorAll<HTMLIFrameElement>('iframe');
+		return Array.from(iframes).find(
+			(iframe) => iframe.contentWindow === messageEventSouce,
+		);
```

Comment:
Do we still need the other strategies? i.e. the ones relying on `slotId` or `iframeId`. Feels like a good opportunity to simplify this method if not!

Reply:
Yes good call! As we discussed in dev time rather than refactor this messenger the consensus was that it's probably going to easier/better to migrate to a new simplified messenger which has none of the cruft that has built up over the years.

Then we can hopefully migrate individual templates/creatives/listeners over with confidence that we don't break anything else.